x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/transport/CrossClusterSecurity.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.security.transport;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.common.settings.ClusterSettings;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.util.concurrent.ConcurrentCollections;
+import org.elasticsearch.transport.TcpTransport;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.TreeMap;
+import java.util.TreeSet;
+
+import static org.elasticsearch.transport.RemoteClusterService.REMOTE_CLUSTER_AUTHORIZATION;
+
+/**
+ * Load/reload API Keys from cluster settings.
+ * Injected into SecurityServerTransportInterceptor.
+ * Latest API Key value needs to be used in header when sending TransportRequest to remote cluster nodes.
+ */
+public class CrossClusterSecurity {
```

Comment:
Still getting up to speed here, but I wonder if it makes sense to store the API key on `RemoteConnection` instead, do the listening inside `RemoteClusterService`, and inject `RemoteClusterService` into the interceptor. Then we could get an API key with something like `remoteClusterService.getRemoteClusterConnection(clusterAlias).getApiKey()` inside the interceptor. 

That would avoid some of the redundancy we're introducing here and keep things consolidated within `RemoteClusterService` and `RemoteConnection`.

Definitely just thinking out loud though.

Reply:
Alternatively, we could extend `RemoteClusterAware` and override the `listenForUpdates` setting

Reply:
This is still a draft PR, so still in progress.

Reply:
Good point. There are other places we can possible do the load/reload, but a consideration is placement of security code in the x-pack Security plug-in. `SecurityServerTransportInterceptor` is in the Security plug-in, `RemoteClusterService` and `RemoteClusterConnection` in server.

Reply:
In terms of dependencies, I don't think there's a problem with `x-pack` relying on `server`. We have that dependency either way because of `REMOTE_CLUSTER_AUTHORIZATION`. 

We could keep the class in x-pack but still extend `RemoteClusterAware`.  There are several examples of this already (e.g., https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/RemoteClusterResolver.java).